api/azure-pipelines.ts

```diff
@@ -1,42 +1,130 @@
+  return await azureDevOpsApiResponse(org, project, `build/latest/${definition}?api-version=${getApiVersion(true)}&branchName=${branch}`)
+}
+
+async function getLatestRelease ({ org, project, definition}: PathArgs) {
+  return await azureDevOpsApiResponse(org, project, `release/releases?api-version=${getApiVersion(true)}&definitionId=${definition}&$top=1`, true)
+}
+
+async function getDeployedRelease ({ org, project, definition, environment}: PathArgs) {
+  return await azureDevOpsApiResponse(org, project, `release/deployments?api-version=${getApiVersion(true)}&definitionId=${definition}&$top=1&deploymentStatus=succeeded&definitionEnvironmentId=${environment}`, true)
+}
+
+async function getTestResultByBuild ({ org, project, build, minLastUpdatedDate, maxLastUpdatedDate}: PathArgs) {
+  return await azureDevOpsApiResponse(org, project, `test/runs?api-version=${getApiVersion(true)}&$top=1&buildIds=${build}&minLastUpdatedDate=${minLastUpdatedDate}&maxLastUpdatedDate=${maxLastUpdatedDate}`)
+}
+
+async function buildStatus ({ org, project, definition, branch = 'master'}: PathArgs) {
+  const response = await getLatestBuild({org, project, definition, branch})
+  const status = statuses[response.result]
+  const color = colors[response.result]
+  return {
+    subject: 'Build',
+    status,
+    color
   }
+}
+
+async function buildVersion ({ org, project, definition, branch = 'master'}: PathArgs) {
+  const response = await getLatestBuild({org, project, definition, branch})
+  const version = response.buildNumber
+  const color = colors[response.result]
+  return {
+    subject: 'Build Version',
+    status: version,
+    color
+  }
+}
 
-  const $ = cheerio.load(response.body)
-  const status = $('g[font-family] > text:nth-child(3)').text()
-  const color = {
-    'succeeded': 'green',
-    'partially succeeded': 'yellow',
-    'failed': 'red'
-  }[status]
+async function buildTestResult ({ org, project, definition, branch = 'master'}: PathArgs) {
+  const build = await getLatestBuild({org, project, definition, branch})
+  const testResult = await getTestResultByBuild({org, project, build: build.id, minLastUpdatedDate: build.startTime, maxLastUpdatedDate: build.finishTime})
+  const runStatistics = testResult.value[0].runStatistics
+  const total: number = testResult.value[0].totalTests
+  const passed: {outcome: string, count: number} = runStatistics.find( (value: { outcome: string; }) => value.outcome === 'Passed')
+  const notExecuted: {outcome: string, count: number} = runStatistics.find( (value: { outcome: string; }) => value.outcome === 'NotExecuted')
+  const failed: {outcome: string, count: number} = runStatistics.find( (value: { outcome: string; }) => value.outcome === 'Failed')
+  
+  const passedCount = !isNullOrUndefined(passed) ? passed.count : 0
+  const notExecutedCount = !isNullOrUndefined(notExecuted) ? notExecuted.count : 0
+  const failedCount =  !isNullOrUndefined(failed) ? failed.count : total - passedCount - notExecutedCount
```

Comment:
We are using typescript ^3.7 now, these can be simplified to:

```js
  const passedCount = passed?.count ?? 0
  const notExecutedCount = notExecuted?.count ?? 0
  const failedCount = failed?.count ?? total - passedCount - notExecutedCount
```

Reply:
Thanks for the suggestion. Implemented the same.